tests/test_dump.py

```diff
@@ -351,6 +351,23 @@ class UserSchema(Schema):
     }
 
 
+def test_field_subclass():
+    """JSON schema generation should not fail on sublcass marshmallow field."""
+
+    class CustomString(fields.String):
+        def __init__(self, *args, **kwargs):
+            super().__init__(**kwargs)
+
+    class TestSchema(Schema):
+        myfield = CustomString()
+
+    schema = TestSchema()
+    try:
+        _ = validate_and_dump(schema)
+    except UnsupportedValueError:
+        pytest.fail("JSON schema generation raised UnsupportedValueError")
```

Comment:
Hmm.. Does this test actually test anything that isn't already covered in virtually every other test case?

I'd expect a test to verify that `UnsupportedValueError` _does_ get thrown with `pytest.raises()` as opposed to a test that is guaranteed to always pass when any other test passes

Reply:
Maybe a test to make sure that `_get_pytype` raises `UnsupportedValueError` would make more sense here?